plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -375,6 +389,37 @@ public function capture_authorized_payment( $order ) {
 		}
 	}
 
+	/**
+	 * Get the authorization ID for the PayPal payment.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return string|null
+	 */
+	private function get_authorization_id( $order ) {
+		$paypal_order_id  = $order->get_meta( '_paypal_order_id', true );
+		$authorization_id = $order->get_meta( '_paypal_authorization_id', true );
+
+		// If the authorization ID is not found, try to retrieve it from the PayPal order details as a fallback for backwards compatibility.
+		if ( ! $authorization_id ) {
+			WC_Gateway_Paypal::log( 'Authorization ID not found, trying to retrieve from PayPal order details as a fallback for backwards compatibility. Order ID: ' . $order->get_id() );
+
+			$order_data         = $this->get_paypal_order_details( $paypal_order_id );
+			$authorization_data = $order_data['purchase_units'][0]['payments']['authorizations'][0] ?? null;
+
+			if ( $authorization_data && isset( $authorization_data['status'] ) && WC_Gateway_Paypal_Constants::STATUS_CAPTURED !== $authorization_data['status'] ) {
+				$authorization_id = $authorization_data['id'];
+				$order->update_meta_data( '_paypal_authorization_id', $authorization_id );
+				$order->update_meta_data( '_paypal_status', WC_Gateway_Paypal_Constants::STATUS_AUTHORIZED );
+				$order->save();
+			} else {
```

Comment:
This logic seems to have too much logic in the initial condition where we get to the `else` and report no authorization ID even in cases where we got authorization data but the status is captured. I think we should update our local data regardless of the status, but possibly validate or normalize the incoming status value.
```suggestion
			if ( $authorization_data && && isset( $authorization_data['id'] ) && isset( $authorization_data['status'] ) ) {
				$authorization_id = $authorization_data['id'];
				$order->update_meta_data( '_paypal_authorization_id', $authorization_id );
				// This direct set may not be safe, but I am not sure what validation would need to look like to get an expected or supported value
				$order->update_meta_data( '_paypal_status', $authorization_data['status'] );
				$order->save();
			} else {
```

Reply:
> his direct set may not be safe, but I am not sure what validation would need to look like to get an expected or supported value

The direct status does not match our mapping. The status in order meta should be `Authorized` to be aligned with our mapping.  

The non-empty  value of `$authorization_data` means that this payment is supposed to go through auth flow (not immediate capture). The `$authorization_data['status']` can have different values (probably), but the value will be `CAPTURED` when the payment have been captured already.

Reply:
I will make the initial condition simpler as per the suggestion.